espnet/optimizer/adaptor.py

```diff
@@ -0,0 +1,106 @@
+"""Import optimizer class dynamically."""
+import argparse
+
+from espnet.optimizer.parser import OPTIMIZER_PARSER_DICT
+from espnet.utils.dynamic_import import dynamic_import
+from espnet.utils.fill_missing_args import fill_missing_args
+
+
+class OptimizerAdaptorInterface:
```

Comment:
According to #943, we decided to use `OptimizerFactoryInterface`.
Why did you change your mind?

Reply:
And I'm still not clear what this class is used for.
Could you tell me how to use `OptimizerAdaptorInterface` and `FunctionalOptimizerAdaptorInterface` properly?

Reply:
Oh, I almost forget it. I will use `OptimizerFactoryInterface` instead of  `OptimizerAdaptorInterface`. First, I planned to use `OptimizerXXXInterface` as usual (or seen in #943) but I jjust tried the functional way is much easier to define the parsers and chainer/pytorch adaptors for checking which one is better. I have no answer yet. Do you think which one is better?

Reply:
Another reason why I used a functional way is that #943 only uses `@classmethod` to define `OptimizerFactoryInterface`. That is essentially the same as a pure function because it does not require any states in `self`. And for Python syntax, the function is much simpler than class.

Reply:
@kan-bayashi I finally rewrite everything from functional style to OOP style